### PR TITLE
Feature/update default jmh version

### PR DESCRIPTION
**Pull request by imanushin**
Several items:
1. Update dependent components to newer version (gradle runner inclusive)
2. Use full gradle distribution (to see more hints in IntelliJ Idea)
3. Update default version of JMH as 1.19 (as written in documentation)

### CODE REVIEW

**Comment by imanushin** [comment]
@melix , would you like to merge or reject my fixes?

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.3-rc-2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4.1-all.zip
```
Comment (melix) [inline gradle/wrapper/gradle-wrapper.properties]: Can you use the normal distribution, please?

Comment (imanushin) [reply gradle/wrapper/gradle-wrapper.properties]: Ok, will do.

Full distribution was recomended by IntelliJ Idea - https://discuss.gradle.org/t/configuring-gradle-wrapper-to-used-distribution-with-sources-in-intellij-ide/5866

**Comment by imanushin** [comment]
Fixed.

Full distribution is recomended by IntelliJ Idea - https://discuss.gradle.org/t/configuring-gradle-wrapper-to-used-distribution-with-sources-in-intellij-ide/5866

**Comment by melix** [comment]
Thanks. I know IntelliJ recommends the fat version, but I disagree. They should download the sources instead.

**Comment by melix** [comment]
There's a build error I need to investigate before I can merge this.

**Comment by imanushin** [comment]
@melix , I agree with you.

> Thanks. I know IntelliJ recommends the fat version, but I disagree. They should download the sources instead.

**Comment by imanushin** [comment]
@melix , could you please take a look on this pull request ?

**Pull request merged**